subctl/issue-182-subctl-a-kubectl-plugin.md

```diff
@@ -0,0 +1,108 @@
+
+A `kubectl-subm` binary would be created and its krew manifest file written.
+
+## Design
+
+To generate `kubectl-subm` binary, the current [`dist/subctl-%.tar.xz`](https://github.com/submariner-io/subctl/blob/devel/Makefile#L92)
+target in subctl repository will be edited to also create `kubectl-subm*.tar.gz` binaries and their checksum file. This is what [Krew 
+expects](https://krew.sigs.k8s.io/docs/developer-guide/plugin-manifest/#sample-plugin-manifest).
+
+    dist/subctl-%: cmd/bin/subctl-%
+	mkdir -p dist
+	tar -cJf $@.tar.xz --transform "s/^cmd.bin/subctl-$(VERSION)/" $<
+    tar -cJf $@.tar.gz --transform "s/^cmd.bin.subctl-$(VERSION)/kubectl-subm-$(VERSION)/" $<
+    md5sum $@.tar.gz > dist/kubectl-subm-$(VERSION)-checksums.txt
+
+The above logic will create `kubectl-subm-$VERSION` binaries under the parent folder itself.
+
+Another option is to drop `.xz` extension files and only keep `.gz` extension files. The effect of this change to other projects would 
+need to be considered, if any.
+
+The [root command name](https://github.com/submariner-io/subctl/blob/devel/cmd/subctl/root.go#L58) would need to be made variable based on
+the argument provided. This would allow running `kubectl-subm` command for the plugin and `subctl` command for standalone `subctl` binary.
+
+    // rootCmd represents the base command when called without any subcommands.
+    var rootCmd = &cobra.Command{
+    Use:   filepath.Base(os.Args[0]),
+    Short: "An installer for Submariner",
+    }
+
+Once `.tar.gz` files are available, krew plugin manifest for subctl would be written as per
+[these instructions](https://krew.sigs.k8s.io/docs/developer-guide/plugin-manifest/#sample-plugin-manifest).
+
+## Usage
+
+Once this is done, users would be able to install subctl via krew. Few of krew command examples are:
+
+    kubectl krew install subm
+    kubectl krew upgrade subm
+
+and use `subctl` as `kubectl-subm` or `kubectl subm`.
+
+    $ kubectl-subctl          
+    An installer for Submariner
+    
+    Usage:
+    kubectl-subctl [command]
+    
+    Available Commands:
+    benchmark           Benchmark tests
+    cloud               Cloud operations
+    completion          Generate the autocompletion script for the specified shell
+    deploy-broker       Deploys the broker
+    diagnose            Run diagnostic checks on the Submariner deployment and report any issues
+    export              Exports a resource to other clusters
+    gather              Gather troubleshooting information from a cluster
+    help                Help about any command
+    join                Connect a cluster to an existing broker
+    recover-broker-info Recovers the broker-info.subm file from the installed Broker
+    show                Show information about Submariner
+    unexport            Stop a resource from being exported to other clusters
+    uninstall           Uninstall Submariner and its components
+    upgrade             Upgrades Submariner
```

Comment:
Upgrade is currently not supported right?

Reply:
Oops. Yes. I had a local copy of upgrade. The plan is to support it in this release though.